Character.cpp

```diff
@@ -87,7 +87,16 @@ int Character::takeDamage(int damage)
 }
 
 
-// #include <assert>
+void Character::levelUpStat(int& stat, std::unique_ptr<int>& initialStat)
+{
+    if ( *initialStat > stat )
+    {
+        stat = *initialStat;
+    } 
+    stat *= 1.1;
+    *initialStat = stat;
+}
```

Comment:
if your second parameter is an `int&`, then you can eliminate all of the pointer dereferencing.
perform the pointer dereferencing when you call this function:
`levelUpStat(armor, *initialArmorLevel);`
- [ ] revise